aws/data_source_aws_iam_role.go

```diff
@@ -27,41 +28,55 @@ func dataSourceAwsIAMRole() *schema.Resource {
 				Computed: true,
 			},
 			"role_id": {
+				Type:       schema.TypeString,
+				Computed:   true,
+				Deprecated: "Use `unique_id` instead",
+			},
+			"unique_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
 			"role_name": {
+				Type:       schema.TypeString,
+				Optional:   true,
+				Deprecated: "Use `name` instead",
+			},
+			"name": {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"create_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
 		},
 	}
 }
 
 func dataSourceAwsIAMRoleRead(d *schema.ResourceData, meta interface{}) error {
-	iamconn := meta.(*AWSClient).iamconn
+	name, hasName := d.GetOk("name")
+	roleName, hasRoleName := d.GetOk("role_name")
 
-	roleName := d.Get("role_name").(string)
-
-	req := &iam.GetRoleInput{
-		RoleName: aws.String(roleName),
+	if !hasName && !hasRoleName {
+		return errwrap.Wrapf("`name` must be set", nil)
```

Comment:
When looking at `Wrapf`, it states:
> Wrapf wraps an error with a formatting message. This is similar to using `fmt.Errorf` to wrap an error. If you're using `fmt.Errorf` to wrap errors, you should replace it with this.

Find it better to use it than `fmt.Errorf`, but `errors.New` could still do it. Do you prefer one or the other?